tools/merge-shards/merge-shards.sh

```diff
@@ -0,0 +1,138 @@
+#!/bin/bash
+# The purpose of this script is a one time data migration from all of our 128
+# schemas into the public schema.
+
+set -euo pipefail
+
+usage() {
+  echo "Usage: merge-shards.sh <SERVER> [migrate]"
+  echo "Valid servers are: jobsrv, originsrv, sessionsrv"
+  exit 1
+}
+
+if [ -z "${PGPASSWORD:-}" ]; then
+  echo "Please ensure that PGPASSWORD is exported in your environment and run this script again."
+  exit 1
+fi
+
+echo "This script will take all of the data that's currently spread across 127 shards and consolidate all of that down into the public Postgres schema."
+echo "Please ensure that you have completed a full database backup before running this script."
+echo
+
+read -rp "Do you want to proceed? [y/n] " answer
+clean_answer=$(echo "$answer" | xargs | tr '[:upper:]' '[:lower:]')
+
+if [ "${clean_answer::1}" != "y" ]; then
```

Comment:
I'm not sure what `${clean_answer::1}` does and couldn't find it in any [documentation](http://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02). Is that a typo?

You could also, do this and obviate the `clean_answer` line:
```bash
if [ "${answer}" != [yY] ]; then
```
Or if you wanted to accept anything that *starts* with y:
```bash
if [ "${answer}" != [yY]* ]; then
```

Reply:
`${clean_answer::1}` is substring extraction.  That was my way of getting the first character of `$clean_answer`.

Reply:
Ah, ok. I get it now, thank you. I prefer `[ "${answer}" != [yY]* ]` to do the same thing, but that's a matter of preference, so go with whatever you prefer.